include/workaround.h

```diff
@@ -11,6 +11,15 @@
 #define CELERITY_WORKAROUND_DPCPP 0
 #endif
 
+#if defined(CELERITY_SIMSYCL)
+#define CELERITY_WORKAROUND_SIMSYCL 1
+#define CELERITY_WORKAROUND_VERSION_MAJOR SIMSYCL_VERSION_MAJOR
+#define CELERITY_WORKAROUND_VERSION_MINOR SIMSYCL_VERSION_MINOR
+#define CELERITY_WORKAROUND_VERSION_PATCH SIMSYCL_VERSION_PATCH
+#else
+#define CELERITY_WORKAROUND_SIMSYCL 0
+#endif
+
```

Comment:
I might have missed it, but are any of these actually used?

If not, I wouldn't include them.

Reply:
It just feels a bit depressing to me to already implement machinery for manual workarounds when none are needed right now.
(Particularly in the case of SimSYCL, where one would hope that it would never need workarounds)

Reply:
Alright, I got rid of them. Also none of the version-specific workaround macros are currently used either because CCPP was the only SYCL implementation we actually upgraded in full-version steps.